### PR TITLE
Problem: walletconnect 2.0 not supporting metamask android (fix #462)

### LINKED ISSUE #462: Problem: walletconnect 2.0 not supporting metamask android
support metamask android for wc 2.0

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: fix send transaction

### CODE REVIEW

**Comment by leejw51crypto** [comment]
fixing audit issue

need to upgrade ethers-rs to 2.0.11
working on 
https://github.com/crypto-com/defi-wallet-core-rs/issues/941

**Comment by leejw51crypto** [comment]
checking ci

**Comment by leejw51crypto** [comment]
fixing WalletConnectRust security issue

File: wallet-connect/examples/web3_v2.rs [hunk 1]
```diff
@@ -14,6 +16,12 @@ use defi_wallet_connect::v2::{Client, ClientOptions, Metadata, RequiredNamespace
 use std::error::Error;
 use std::io::BufRead;
 
+const RPC_URL: &str = env!("MY_RPC_URL");
+// convert string to u64
```
Comment (rarcifa) [inline wallet-connect/examples/web3_v2.rs]: comment can be removed

File: wallet-connect/examples/web3_v2.rs [hunk 2]
```diff
@@ -197,6 +215,39 @@ async fn make_qrcode(uri: &str) -> Result<()> {
 
     Ok(())
 }
+
+// Function to create, sign, and send an Ethereum transaction
```
Comment (rarcifa) [inline wallet-connect/examples/web3_v2.rs:218]: Have you ever used Rustdoc for documentation? Its similar to doxygen and jsdocs. This is just an idea for the future :) 

Example

```rs
/// Asynchronously sends an Ethereum transaction.
///
/// This function constructs and sends an Ethereum transaction using EIP-1559 for fee calculation. 
/// It takes the sending and receiving addresses, along with the amount to be sent, and returns the transaction hash.
///
/// # Arguments
/// * `client` - A mutable reference to the Ethereum `Client`.
/// * `from` - The sender's Ethereum address in hexadecimal string format.
/// * `to` - The receiver's Ethereum address in hexadecimal string format.
/// * `amount` - The amount of Ethereum (in Wei) to be sent, encapsulated in a `U256`.
///
/// # Returns
/// A `Result` containing the transaction hash (`TxHash`) on success.
///
/// # Errors
/// Returns an error if any of the following occurs:
/// - Invalid format of `from` or `to` address.
/// - Issues connecting to the Ethereum provider.
/// - Failure in constructing or sending the transaction.
///
/// # Example
/// ```
/// # use ethers::prelude::*;
/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
/// let mut client = ...; // Initialize Ethereum client
/// let from = "0x...";
/// let to = "0x...";
/// let amount = U256::from(1000000000000000000u64); // 1 ETH in Wei
///
/// let tx_hash = send_ethereum_transaction(&mut client, from, to, amount).await?;
/// println!("Transaction sent with hash: {:?}", tx_hash);
/// # Ok(())
/// # }
/// ```
///
/// # Notes
/// - The transaction uses EIP-1559 for fee calculation.
/// - The `gas`, `max_priority_fee_per_gas`, and `max_fee_per_gas` values are hard-coded in this function.
/// - The `chain_id`, `nonce`, and other transaction parameters are obtained at runtime.
 ```

File: wallet-connect/examples/web3_v2.rs [hunk 3]
```diff
@@ -237,37 +289,34 @@ async fn main() -> Result<(), Box<dyn Error>> {
...
-    if test_tx {
+    if test_sign_tx {
         // read env MYTOADDRESS
-        let to = std::env::var("MYTOADDRESS").expect("MYTOADDRESS not set");
-        let fromaddress = namespaces.get_ethereum_addresses()[0].address.clone();
-        // print fromaddress
-        println!("fromaddress= {:?}", fromaddress);
+        let fromaddress = FROM_ADDRESS.to_string();
+        let toaddress = TO_ADDRESS.to_string();
+        // convert fromaddress to 20 fixed byte array
```
Comment (rarcifa) [inline wallet-connect/examples/web3_v2.rs:299]: not needed

File: wallet-connect/src/v2/crypto.rs [hunk 4]
```diff
@@ -80,7 +80,12 @@ mod test {
             132, 149, 158, 189, 217, 78, 224, 11, 145, 159, 235, 198, 115,
         ];
         let key = Key::from_raw(dapp_secret);
-        let Some((topic, _)) = derive_symkey_topic("f22533e8a398c465569c04c14b853c86b63ad94ffa916861eb138819c8be475f", &key) else { panic!("can't derive topic") };
+        let Some((topic, _)) = derive_symkey_topic(
```
Comment (rarcifa) [inline wallet-connect/src/v2/crypto.rs]: maybe we can set the TOPIC in a const and then add the TOPIC variable like so:

```
const TOPIC: &str = "f22533e8a398c465569c04c14b853c86b63ad94ffa916861eb138819c8be475f";

// ... later in your code ...
let Some((topic, _)) = derive_symkey_topic(TOPIC, &key) else {
    panic!("can't derive topic");
};
```

**Review by rarcifa — APPROVED** [review]
LGTM. left a few comments

File: wallet-connect/src/v2/crypto.rs (hunk 4 above)
Comment (leejw51crypto) [reply wallet-connect/src/v2/crypto.rs]: ok

File: wallet-connect/examples/web3_v2.rs (hunk 3 above)
Comment (leejw51crypto) [reply wallet-connect/examples/web3_v2.rs:299]: ok

File: wallet-connect/examples/web3_v2.rs (hunk 1 above)
Comment (leejw51crypto) [reply wallet-connect/examples/web3_v2.rs]: i'll check

**Pull request merged**